screens/changePassword/ChangePasswordScreen.tsx

```diff
@@ -1,68 +1,131 @@
 import { BodyText, Headline } from "@components/Text"
+import { PasswordTextField } from "@components/TextFields"
 import { PrimaryButton } from "@components/common/Buttons"
 import { AppStyles } from "@lib/AppColorStyle"
-import { validateNewPassword } from "@lib/Password"
+import { Password } from "@lib/Password"
 import { Auth } from "aws-amplify"
 import React, { useState } from "react"
-import { SafeAreaView, ScrollView, StyleSheet, View } from "react-native"
+import { Alert, SafeAreaView, ScrollView, StyleSheet, View } from "react-native"
 import { TouchableOpacity } from "react-native-gesture-handler"
-import { PasswordField } from "./PasswordField"
+import { useMutation } from "react-query"
 
-let isValidForm = false
-const oldAccountPassword: string = "Icarus43$"
+const isValidForm = false
 
 type ChangePasswordProps = {
-  onPasswordChangeSubmitted?: (
-    oldPass: string,
-    newPass: string
-  ) => Promise<boolean>
+  onPasswordChangeSubmitted: (
+    uncheckedOldPass: string,
+    newPass: Password
+  ) => Promise<ChangePasswordSubmissionResult>
 }
 
-const changePassword = async (oldPass: string, newPass: string) => {
+type ChangePasswordSubmission =
+  | { status: "valid"; submit: () => void }
+  | {
+      status: "invalid"
+      submit?: undefined
+      error: ChangePasswordErrorReason
+    }
+  | { status: "submitting" }
+
+type ChangePasswordErrorReason =
+  | "current-matches-new"
+  | "reenter-does-not-match-new"
+  | "weak-new-password"
+  | "incorrect-current-password"
+
+type ChangePasswordSubmissionResult = "valid" | "invalid" | "incorrect-password"
+
+export const changePassword = async (
+  uncheckedOldPass: string,
+  newPass: Password
+) => {
   return await Auth.currentAuthenticatedUser()
     .then((user) => {
-      return Auth.changePassword(user, oldPass, newPass)
+      return Auth.changePassword(user, uncheckedOldPass, newPass.rawValue)
     })
     .then((data) => true)
     .catch((err) => false)
 }
 
+export const useChangePassword = ({
+  onPasswordChangeSubmitted
+}: ChangePasswordProps) => {
+  const [currentPassword, setCurrentPassword] = useState("")
```

Comment:
Create a multi-setter for the 3 states, instead of having 3 useStates, to allow for just one function to be accounted for instead of 3 states